tests/admin_views/test_related_object_lookups.py

```diff
@@ -117,3 +117,46 @@ def test_related_object_update_with_camel_casing(self):
             <option value="1">{interesting_name}</option>
             """,
         )
+
+    def test_add_related_entry_from_m2m_updates_lists(self):
+        from selenium.webdriver.common.by import By
+
+        def _get_HTML_inside_element_by_id(id_):
+            return self.selenium.find_element(By.ID, id_).get_attribute("innerHTML")
+
+        add_url = reverse("admin:admin_views_camelcaserelatedmodel_add")
+        self.selenium.get(self.live_server_url + add_url)
+        name = "Bergeron"
+
+        # Add a Topping from the plus button on the m2m field
+        self.selenium.find_element(By.ID, "add_id_m2m").click()
+
+        # Switch to the popup window
+        self.wait_for_and_switch_to_popup()
+
+        # Find the name field and add text to it
+        self.selenium.find_element(By.ID, "id_interesting_name").send_keys(name)
+
+        # Now find the save button and click it
+        self.selenium.find_element(By.NAME, "_save").click()
+
+        # Return to the main window
+        self.wait_until(lambda d: len(d.window_handles) == 1, 1)
+        self.selenium.switch_to.window(self.selenium.window_handles[0])
+
+        # Check that the state is in the Available Source box
+        self.assertHTMLEqual(
+            _get_HTML_inside_element_by_id("id_m2m_from"),
+            f"""
+            <option selected value="1">{name}</option>
+            """,
+        )
+
+        # Now check that the state shows in the fk box
```

Comment:
I think these comments are not needed 👍 
```suggestion
        self.selenium.find_element(By.ID, "add_id_m2m").click()
        self.wait_for_and_switch_to_popup()

        self.selenium.find_element(By.ID, "id_interesting_name").send_keys(name)
        self.selenium.find_element(By.NAME, "_save").click()

        self.wait_until(lambda d: len(d.window_handles) == 1, 1)
        self.selenium.switch_to.window(self.selenium.window_handles[0])

        self.assertHTMLEqual(
            _get_HTML_inside_element_by_id("id_m2m_from"),
            f"""
            <option selected value="1">{name}</option>
            """,
        )
```